packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -204,9 +225,89 @@ const reducer: Reducer<State> = (state = INITIAL_STATE, action) => {
   return state;
 };
 
+const confirmWriteOperationIfNeeded = async (
+  instance: MongoDBInstance,
+  dataService: DataService,
+  namespace: string,
+  pipeline: Document[]
+) => {
+  if (!pipeline.length) {
+    return true;
+  }
+
+  const lastStageOperator = getStageOperator(pipeline[pipeline.length - 1]);
```

Comment:
Just FYI, `getStageOperator` supports nullish values, so you can just do one check for `lastStageOperator` below instead of checking the array length first
```suggestion
  const lastStageOperator = getStageOperator(pipeline[pipeline.length - 1]);
```